### ISSUE TITLE
[question] if COLMAP gives significantly better results...

### ISSUE THREAD

**Issue by skinkie** [alicevision/Meshroom#2126]
**Describe the problem**
I have been trying the new 2023 release and my results are very poor. Poor as in: recovery basically leads to a flat pointcloud, almost as a 360-panorama is being created. I have tried to change some settings including different initial pairs, all to no significant improvement.

I have naively thrown the all my images into COLMAP (never used it before).

**Screenshots**
![image](https://github.com/alicevision/Meshroom/assets/502394/cd8071ed-6b1d-4365-9d95-5c75bbccbe46)

![image](https://github.com/alicevision/Meshroom/assets/502394/d10e3fa9-8443-41cd-9dd0-ca14de0548ed)

**Dataset**
If someone wants to try it out, happy to share the dataset. But maybe more people have the issue with the latest release.

**Comment by fabiencastan** [comment]
Thanks for the report. I would be interested to analyze your dataset. Have you already tried with an older version?

**Comment by skinkie** [comment]
> Thanks for the report. I would be interested to analyze your dataset. Have you already tried with an older version?

Not yet. Especially because I have always issues with Cuda, so I was quite happy that this release actually worked on my system, without going into a night of battling against dependencies. http://download.stefan.konink.de/meshroom-bugs/all-images.tar

**Comment by msanta** [comment]
I have used @skinkie's dataset with 2023.2 and 2021.1 using default settings. Here are the results:

2023.2
![image](https://github.com/alicevision/Meshroom/assets/5391990/24486ada-ae86-44cd-8b33-730f01bd11ad)


2021.1
![image](https://github.com/alicevision/Meshroom/assets/5391990/b434d4df-1aec-4a57-829b-7d96090252af)

**Comment by balaclava9** [comment]
yes COLMAP was able to find all camera positions in my 15 image data set, while meshroom could not.

**Comment by tak-ho** [comment]
Is there any problem with 2023.2? Could I adjust some settings to produce 2021.1 result with 2023.2 or I could only go back to 2021.1?  Thanks!

**Comment by servantftechnicolor** [comment]
Please update the intrinsics in camerainit to use 22 as focal length and -1 to default focal length.

![image](https://github.com/alicevision/Meshroom/assets/100348063/ebfd2810-9899-4a72-a523-45118b2c85e2)

- Your device is not recognized as it is not part of the sensors db so the sensor width is defaulted to 36mm
- Your images are giving a focal length of 3.5mm (which for a sensor width 36mm, is a *very* small values ; 2 * atan2(36/2, 3.5) ~=160°)

I assume some default logic changed between the two versions; but anyway, always make sure your device is in the sensor database.

**Comment by skinkie** [comment]
I remember I have done a pull request to add the Nokia 3.1 Plus. This obviously works well for photo's with EXIF, but not for video. I wonder, what did you do to determine that it was 22 as focal length?

**Comment by servantftechnicolor** [comment]
and the result i got : 

![image](https://github.com/alicevision/Meshroom/assets/100348063/163ac887-dc4d-40a4-86c4-84c11640de95)

I found default fov on the web

tan(fov/2) = (36/2) / focal ... it's just an approximation. That's why i set "default focal length" to -1, to make sure the structure from motion does not enforce this focal length but only use it as some starting point.

**Comment by skinkie** [comment]
Based on the colour of the camera's you did not share intrinsics?

**Comment by servantftechnicolor** [comment]
I don't understand your sentence. There are two intrinsics in the scene.

**Comment by skinkie** [comment]
@servantftechnicolor If you review the different camera positions you can see they all differ from colour. If I recall correctly, this means the intrinsics of the cameras are all different, since it was the same device, that should not be the case.

**Comment by servantftechnicolor** [comment]
I don't think so. It's a random color.

**Comment by fabiencastan** [comment]
The camera gizmo's color is unique per view. It facilitates visualization when multiple cameras are close one to the other.

**Comment by fabiencastan** [comment]
You can see on the number of intrinsics here:
![image](https://github.com/alicevision/Meshroom/assets/153585/5bea16dd-997b-48d8-a97a-8b944c3af312)
Or the detailed list of intrinsics on the CameraInit node.

**Comment by skinkie** [comment]
@servantftechnicolor 

![image](https://github.com/alicevision/Meshroom/assets/502394/f65c13a9-bd60-46e8-b726-1f15b6fcb91a)

**Comment by msanta** [comment]
@skinkie Did you set the `initial focal length` and `focal length` values for both of the intrinsics?

**Comment by skinkie** [comment]
> @skinkie Did you set the `initial focal length` and `focal length` values for both of the intrinsics?

I did not. When I did set them to -1, and 22 for the second part. My results are _better_, but not at the expected quality. I have been using sift instead of dspsift (so GPU acceleration works) and voctree. As you can see from the image below the camera positions outside have not been reconstructed. (339/403)

![image](https://github.com/alicevision/Meshroom/assets/502394/58652077-e0e3-4fc2-b15c-ba239b01042a)
